### PR TITLE
docs: note that changing encryption option settings for existing clusters is not dangerous

### PR DESCRIPTION
**Pull request by pluser**
We have prohibited changing all network-related configurations after cluster creation. However, as a result of #13584, we found that the connections.encryption flag can be regarded as an exception.

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [x] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [x] Documentation has been updated, if necessary.
- [x] Unit tests have been added, if necessary.
- [x] Integration tests have been added, if necessary.

### CODE REVIEW

**Review by satoru-takeuchi — APPROVED** [review]

**Review by subhamkrai — APPROVED** [review]
lgtm

File: Documentation/CRDs/Cluster/ceph-cluster-crd.md [hunk 1]
```diff
@@ -221,7 +221,7 @@ Configure the network that will be enabled for the cluster and services.
 
 !!! caution
     Changing networking configuration after a Ceph cluster has been deployed is NOT
-    supported and will result in a non-functioning cluster.
+    supported except for encryption settings and will result in a non-functioning cluster.
```
Comment (subhamkrai) [inline Documentation/CRDs/Cluster/ceph-cluster-crd.md:224]:
```suggestion
    Except for encryption settings changing networking configuration after a Ceph cluster has been deployed is *NOT* supported and will result in a non-functioning cluster.
```
what about this?

**Comment by pluser** [comment]
@travisn Could you take a look this PR?

File: Documentation/CRDs/Cluster/ceph-cluster-crd.md [hunk 2]
```diff
@@ -220,7 +220,7 @@ Configure the network that will be enabled for the cluster and services.
       See the kernel requirements above for encryption.
 
 !!! caution
-    Changing networking configuration after a Ceph cluster has been deployed is NOT
+    Except for encryption settings changing networking configuration after a Ceph cluster has been deployed is *NOT*
```
Comment (travisn) [inline Documentation/CRDs/Cluster/ceph-cluster-crd.md:223]: The wording is tricky. How about this?
```suggestion
    Changing networking configuration after a Ceph cluster has been deployed is only supported for the network encryption settings. Changing other network settings is *NOT* supported and will result in a non-functioning cluster.
```

**Review by travisn — APPROVED** [review]

**Pull request merged**